src/test/registration.test.tsx

```diff
@@ -0,0 +1,40 @@
+// registration.test.tsx
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import { Registration } from '../pages/registration/registration';
+
+describe('Registration Page', () => {
+  it('renders registration form', () => {
+    const { getByText } = render(<Registration />);
```

Comment:
```suggestion
    const { getByText } = renderInsideRouter(<Registration />);
```

Here, we wrap the ```Registration``` element in a ```MemoryRouter```.

Reply:
this definitely solves the problem